### ISSUE TITLE
To do list

### ISSUE THREAD

**Issue by persianpros** [PLi-metas/pli-extras#2]
https://github.com/PLi-metas/meta-xtrend-extra for extra models
https://github.com/PLi-metas/meta-gigablue-extra for extra models

**Comment by MastaG** [comment]
For the record, I've built a enigma2-plugin-extensions-openwebif-extras package which replaces the enigma2-plugin-extensions-openwebif package.

See: https://github.com/PLi-metas/pli-extras/blob/develop/recipes-openpli/enigma2-plugins/enigma2-plugin-extensions-openwebif-extras.bb

We can easily add extra receivers in here since it was a bitch to maintain it using multiple bbappends in every BSP-layer.

Now for RC-images in enigma2.. we can try to do this using the enigma2.bbappend:
https://github.com/PLi-metas/pli-extras/blob/develop/recipes-openpli/enigma2/enigma2.bbappend

Simply create a big zip-archive which contains a folder names "images" which has a model.png/model.xml for every receiver. And a list names extras.cfg.
Then add something like:

SRC_URI_append += "file://extra_images.zip"

do_configure_prepend() {
mv ${WORKDIR}/images/*.{xml,png} ${S}/data/rc_models/
cat ${WORKDIR}/images/extras.cfg >> ${S}/data/rc_models/rc_models.cfg
}

That way our extra rc_models will get added to enigma2 before configure runs.. and we don't have to fork it.. or deal with annoying patches since they tend to break because PLi adds new rc_models all the time...

**Comment by persianpros** [comment]
100% Agree :)
Would you please do it for the first time so I could follow your example?

**Comment by MastaG** [comment]
Done :)
See: https://github.com/PLi-metas/pli-extras/commit/1330d35fc949c3a6a4df46cf786fc68e32244fd9
And: https://github.com/PLi-metas/extra_rc_models

**Comment by persianpros** [comment]
Well done my friend well done :)
I will add new files next week but it only works for enigma2 not OpenWebIF right?

**Comment by MastaG** [comment]
yes, I'm afraid so.
But for OpenWebIf there's already many contributers see:
https://github.com/E2OpenPlugins/e2openplugin-OpenWebif/tree/master/plugin/public/images/remotes
spark for example is already there.

So if there's already a picture you can simply add them to the list in:
https://github.com/PLi-metas/pli-extras/blob/develop/recipes-openpli/enigma2-plugins/enigma2-plugin-extensions-openwebif-extras.bb

I'll modify the extra_rc_models repo later to include box pictures as well and add the same procedure for OpenWebIf soon.. then you can simply add them to the repo like:
boxes/machine.png
and add something to the recipe like:
('machine', 'machine.png', 'machine.png', 'machine.html'),

Hope to add it soon.

**Comment by persianpros** [comment]
Ok now we have "webif" folder inside https://github.com/PLi-metas/extra_rc_models so how can we support OpenWebIF too?

**Comment by MastaG** [comment]
I still need to update the openwebif-extras.bb recipe for that :)
Will do this later this week.

**Comment by MastaG** [comment]
Done, extra_rc_models is now also used for the OpenWebIf, see: https://github.com/PLi-metas/pli-extras/commit/7c8c7e6907ab4b98549a0a567722e5d9dc158054

extra_rc_models/webif/*.png is copied to OpenWebIf/plugin/public/images/boxes/
extra_rc_models/webif/*.html is copied to OpenWebIf/plugin/public/static/remotes/
extra_rc_models/*.png is copied to OpenWebIf/plugin/public/images/remotes/

Existing images with the same filename will be overwritten.

But in order for them to show up, you'll need to append a line to:
https://github.com/PLi-metas/pli-extras/blob/develop/recipes-openpli/enigma2-plugins/enigma2-plugin-extensions-openwebif-extras.bb
('MACHINE, 'boximage.png', 'remoteimage.png', 'htmlfile.html'),

Note there's spaces in front of the line, not tabs :)

**Comment by persianpros** [comment]
Well done MastaG well done, I always love your work and the way you doing things, You're a great teacher :+1:

**Comment by MastaG** [comment]
Thank you my friend.
I forgot the owibranding file indeed.
It's added now: https://github.com/PLi-metas/pli-extras/commit/341a1b28d280cb5225af7f16d386c1c1d997a9e9

**Comment by persianpros** [comment]
Great, can you make sure "enigma2-plugin-systemplugins-servicemp3" get pre installed for all boxes?

I guess we need something like https://github.com/PLi-metas/meta-dream/blob/develop/recipes-openpli/images/openpli-enigma2-image.bbappend

**Comment by MastaG** [comment]
Isn't it getting pre-installed?
https://github.com/PLi-metas/pli-extras/blob/104b6553f9fe2c43bab421287fd4792e07e194cf/recipes-openpli/images/openpli-enigma2-image.bbappend#L28

**Comment by MastaG** [comment]
It's only installed on non smallflash receivers.
It doesn't fit dm500hd/800hd/800se because of 64MB flash.
serviceapp pulls in exteplayer3, which pulls in the ffmpeg libs.. which are large.

**Comment by MastaG** [comment]
oh wait, you meant servicemp3..lol
It should be there by default using openpli-oe-core:

conf/distro/openpli-common.conf:PREFERRED_PROVIDER_virtual/enigma2-mediaservice ?= "enigma2-plugin-systemplugins-servicemp3"

original enigma2.bb:
RRECOMMENDS_${PN} = " \
	enigma2-plugin-skins-pli-hd \
	hotplug-e2-helper \
	glibc-gconv-utf-16 \
	python-sendfile \
	ofgwrite \
	**virtual/enigma2-mediaservice** \
	"

**Comment by persianpros** [comment]
I'm not sure that is pre-installed or not :|

**Comment by MastaG** [comment]
I have not built for other receivers lately.
But it should be there unless a BSP-layer specifies another media-service.

See: https://github.com/OpenPLi/openpli-oe-core/commit/f3d733e814f69d71983dcb6a7657ddc1a123fa0f

For example for sh4 (meta-fulan), I override PLi's preferred provider like this:
PREFERRED_PROVIDER_virtual/enigma2-mediaservice ?= "enigma2-plugin-systemplugins-servicemp3epl"

There I use Taapats servicemp3epl which has libeplayer3 builtin as well.
But in the end it's PLi's enigma2 which pulls in "virtual/enigma2-mediaservice" using RRECOMMENDS.

So I can think of two possible causes for the missing servicemp3 plugin:
1. There's a BSP-Layer with some enigma2.bbappend, which rewrites RRECOMMENDS_${PN}, thus removing "virtual/enigma2-mediaservice"
2. There's a BSP-Layer which set its own: PREFERRED_PROVIDER_virtual/enigma2-mediaservice

**Comment by persianpros** [comment]
meta-azbox is ready but "azboxhd" is a "smallflash" and all upx things are in "meta-dream" so can "azboxhd" use it or should we move all upx related things from "meta-dream" to "pli-extras"?

**Comment by Hains** [comment]
meta-azbox can use it, but it's a good idea to move all the non meta-dream related stuff (all bbappends) to pli-extras.

**Comment by persianpros** [comment]
+1 if you have time and know where all these recipes please move them to pli-extras as meta-dream is crowded too much :(

**Comment by Hains** [comment]
Done. As you see it's a huge change. I notice a few duplicate bbappends, so i merged them into one.

**Comment by MastaG** [comment]
Thank you Hains,

However, there's now dreambox specific bbappends and patches  in pli-extras.
Shouldn't we keep them in meta-dream instead?
You can have more than one bbappend for a certain recipe.

For example, there's a enigma2.bbappend from meta-dream will add the Dreambox specific patches.
But the enigma2.bbappend from meta-fulan will just set the spark spefic sources with:
SRC_URL_sh4 =..

This means that each package will indeed get this information added to it's enigma2.bbappend.
But since the code only affects spark receivers it doesn't matter.

So in short, each BSP-layer can have it's own bbappend, applying stuff for it's own machines/architecture to the recipe.
As long as it doesn't create conflicts for other receivers/architectures.
That's why there's many base_contains and _MACHINE_FEATURE _ARCH _MACHINE etc..

**Comment by Hains** [comment]
Yes. I was in doubt regarding e.g. glibc.. Yes you'r right, it's dreambox specific.
But the purpose of this is, to keep it clear for everybody.
No duplicate bbappends shredded over the place.

I restore glibc 2.19 to meta-dream.
I also deleted two files of openwebinterface, that explains the difference between the additions(193) and  deletions(195).

**Comment by Hains** [comment]
Also enigma2 patches dreambox.

**Comment by MastaG** [comment]
Thank you Hains, I'll help a bit as well, later this week.
When I'm done modding my xbox 360 gaming console with a glitch chip lol

**Comment by Hains** [comment]
Done..Except exteplayer, it does a dreambox and a non dreambox patch..So?
Cleanup in meta-dream.
All bbappends with inherit upx to pli-extras.
smbnetfs, vsftpd and gnupg as well.
autossh and openwebifinterace removed.

**Comment by persianpros** [comment]
Well done my friend well done :100: Now meta-dream is better than ever :)

**Comment by persianpros** [comment]
azboxhd image is still more than 64MB, what do you think we can remove safely from its image?

**Comment by Hains** [comment]
You can try to compile with Os instead of O2 in the openpli-common.conf file.
That save a few MB. But users want also to install plugins and they want everything(my experience).
So drop 'something' is no option.
Also be ware that when we switch over to rocko branch, the image will be bigger and you have the same problem again.

**Comment by Hains** [comment]
I'll add GPG key to my account these days.

**Comment by Hains** [comment]
Now i have GPG signature verification. 
So now you can use 'git commit -S ' in submodule.sh if you want.

**Comment by persianpros** [comment]
@Hains Would you please do a submodule.sh? My internet speed sucks and I can't access github without ssh :(
